### ISSUE TITLE
Occasional failure in IPAM unit tests

### ISSUE THREAD

**Issue by rade** [weaveworks/weave#1651]
I have seen the following a couple of times...

```
make -k tests 
tools/test
INFO: 2015/11/07 16:43:06.774111 [allocator 01:00:00:02:00:00] Claim 10.0.3.2 for b01df00d: address allocator still initializing; will try later.
INFO: 2015/11/07 16:43:06.777443 [allocator 01:00:00:02:00:00] Ignored address 10.0.4.2 claimed by abcdef - not in our universe
INFO: 2015/11/07 16:43:06.978920 [allocator 01:00:00:01:00:00] Shutdown
Cancelling allocate
--- FAIL: TestHTTPCancel (0.10s)
	Error Trace:	
	Error:		Error: Allocate returned non-nil
		
FAIL
FAIL	github.com/weaveworks/weave/ipam	1.346s
```

**Comment by tomwilkie** [comment]
I've hit this a couple of times recently - https://circleci.com/gh/weaveworks/weave/4271 for one.

**Comment by tomwilkie** [comment]
And again - https://circleci.com/gh/weaveworks/weave/4275

**Comment by bboreham** [comment]
https://github.com/weaveworks/weave/blob/master/ipam/allocator_test.go#L180 seems suspect.

I'll fix it.

**Comment by bboreham** [comment]
Actually that code is correct.  But I've fixed the test anyway.

**Comment by bboreham** [comment]
Fixed by #1862
